resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/UserAdapter.java

```diff
@@ -0,0 +1,249 @@
+            case "created_timestamp":
+                return String.valueOf(entity.getCreatedTimestamp());
+            case "super_user":
+                return String.valueOf(entity.getSuperUser());
+            case "organizations":
+                return entity.getOrganizations() != null ? OrganizationObject.toStringList(entity.getOrganizations())
+                        : "[]";
+            default:
+                return super.getFirstAttribute(name);
+        }
+    }
+
+    @Override
+    public Map<String, List<String>> getAttributes() {
+        log.debug("getAttributes");
+        MultivaluedHashMap<String, String> attrs = new MultivaluedHashMap<>();
+        attrs.add(UserModel.USERNAME, entity.getEmail());
+        attrs.add(UserModel.EMAIL, entity.getEmail() != null ? entity.getEmail() : "");
+        attrs.add(UserModel.FIRST_NAME, entity.getFirstName() != null ? entity.getFirstName() : "");
+        attrs.add(UserModel.LAST_NAME, entity.getLastName() != null ? entity.getLastName() : "");
+        attrs.add("created_timestamp",
+                String.valueOf(entity.getCreatedTimestamp() != null ? entity.getCreatedTimestamp() : ""));
+        attrs.add("super_user",
+                String.valueOf(entity.getSuperUser() != null ? entity.getSuperUser() : ""));
+        attrs.add("organizations",
+                entity.getOrganizations() != null ? OrganizationObject.toStringList(entity.getOrganizations()) : "[]");
+        // Include other attributes as needed
+        return attrs;
+    }
+
+    @Override
+    public Stream<String> getAttributeStream(String name) {
+        log.debug("getAttributeStream: " + name);
+        switch (name) {
+            case UserModel.USERNAME:
+                return Stream.of(entity.getEmail());
+            case UserModel.EMAIL:
+                return entity.getEmail() != null ? Stream.of(entity.getEmail()) : Stream.empty();
+            case UserModel.FIRST_NAME:
+                return entity.getFirstName() != null ? Stream.of(entity.getFirstName()) : Stream.empty();
+            case UserModel.LAST_NAME:
+                return entity.getLastName() != null ? Stream.of(entity.getLastName()) : Stream.empty();
+            case "created_timestamp":
+                return entity.getCreatedTimestamp() != null ? Stream.of(String.valueOf(entity.getCreatedTimestamp()))
+                        : Stream.empty();
+            case "super_user":
+                return entity.getSuperUser() != null ? Stream.of(String.valueOf(entity.getSuperUser()))
+                        : Stream.empty();
+            case "organizations":
+                return entity.getOrganizations() != null
+                        ? Stream.of(OrganizationObject.toStringList(entity.getOrganizations()))
+                        : Stream.empty();
+            default:
+                return super.getAttributeStream(name);
+        }
+    }
+
+    private void save() {
+        provider.updateUser(realm, entity);
+    }
```

Comment:
issue: I understand the desire to encapsulate the "save" functionality into a function like this, but by not taking in parameters (like realm and entity) we're making it harder for future developers to understand what this method does. I like the clarity of `updateUser` already, so could we just use a direct call to `updateUser` instead? The method call indicates what's being updated by showing that we're passing in `realm` and `entity`, so users of the method don't need to wonder what's being updated or how.

Reply:
Updated to be named saveUpdatedUserAttributes and take in the entity to save. I slightly prefer separating it, because it's duplicated and it's a nice abstraction in case we change the updateUser method later. I added a javadoc comment to saveUpdatedUserAttributes describing what it relies on for clarity